src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/ResendStaffNotificationsCoreCaseDataTest.java

```diff
@@ -0,0 +1,250 @@
+        final String event = "claim-issue";
+        given(coreCaseDataApi.searchForCitizen(
+            any(),
+            any(),
+            any(),
+            any(),
+            any(),
+            eq(ImmutableMap.of("case.referenceNumber", CASE_REFERENCE, "page", PAGE, "sortDirection", "desc"))
+            )
+        ).willReturn(listOfCaseDetails());
+        ResultActions result = makeRequest(nonExistingClaimReference, event);
+        assertThat(result).isNotNull();
+        result.andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void shouldRespond404WhenEventIsNotSupported() throws Exception {
+        final String nonExistingEvent = "some-event";
+
+        given(coreCaseDataApi.searchForCitizen(
+            any(),
+            any(),
+            any(),
+            any(),
+            any(),
+            eq(ImmutableMap.of("case.referenceNumber", CASE_REFERENCE, "page", PAGE, "sortDirection", "desc"))
+            )
+        ).willReturn(listOfCaseDetails());
+        ResultActions result = makeRequest(CASE_REFERENCE, nonExistingEvent);
+        assertThat(result).isNotNull();
+        result.andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void shouldRespond409AndNotProceedForClaimIssuedEventWhenClaimIsLinkedToDefendant() throws Exception {
+        final String event = "claim-issued";
+
+        given(coreCaseDataApi.searchForCitizen(
+            any(),
+            any(),
+            any(),
+            any(),
+            any(),
+            eq(ImmutableMap.of("case.referenceNumber", CASE_REFERENCE, "page", PAGE, "sortDirection", "desc"))
+            )
+        ).willReturn(listOfCaseDetailsWithDefendant());
+
+        makeRequest(CASE_REFERENCE, event).andExpect(status().isConflict());
+
+        verify(emailService, never()).sendEmail(any(), any());
+    }
+
+    @Test
+    public void shouldRespond200AndSendNotificationsForClaimIssuedEvent() throws Exception {
+        final String event = "claim-issued";
+        final Claim claim = SampleClaim.builder()
+            .withReferenceNumber(CASE_REFERENCE)
+            .withDefendantId(null).build();
+
+        GeneratePinResponse pinResponse = new GeneratePinResponse("pin-123", "333");
+        given(userService.generatePin(anyString(), eq(AUTHORISATION_TOKEN))).willReturn(pinResponse);
+        given(sendLetterApi.sendLetter(any(), any())).willReturn(new SendLetterResponse(UUID.randomUUID()));
+        given(coreCaseDataApi.searchForCitizen(
+            any(),
+            any(),
+            any(),
+            any(),
+            any(),
+            eq(ImmutableMap.of("case.referenceNumber", CASE_REFERENCE, "page", PAGE, "sortDirection", "desc"))
+            )
+        ).willReturn(listOfCaseDetails());
+
+        makeRequest(claim.getReferenceNumber(), event).andExpect(status().isOk());
+
+        verify(emailService, times(2)).sendEmail(eq("sender@example.com"), emailDataArgument.capture());
```

Comment:
Why do we expect this notification to be sent 2 times?

Reply:
DocumentGeneratedEvent is listened by notifyRobotOfClaimIssue and notifyStaffOfClaimIssue and both uses emailService.sendEmail.